src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/TypeBuilderImpl.cs

```diff
@@ -734,7 +734,7 @@ private static bool MatchesTheFilter(MethodBuilderImpl method, BindingFlags meth
 
                 if (argType.IsArray != paramType.IsArray ||
                     argType.IsByRef != paramType.IsByRef ||
-                    argType.IsPointer != argType.IsPointer)
```

Comment:
This is a real corner case bug, but I do not think it is worth it to add a test case to hit.

Reply:
https://github.com/dotnet/runtime/pull/112048